api/s2n.h

```diff
@@ -687,6 +687,266 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
 S2N_API 
 extern int s2n_cert_get_utf8_string_from_extension_data(const uint8_t *extension_data, uint32_t extension_len, uint8_t *out_data, uint32_t *out_len);
 
+/* Pre-shared key (PSK) Hash Algorithm - RFC 8446 Section-2.2 */
+typedef enum {
+    S2N_PSK_HMAC_SHA256,
+    S2N_PSK_HMAC_SHA384,
+} s2n_psk_hmac;
+
+struct s2n_psk;
+
+/**
+ * Creates a new s2n external Pre-shared key (PSK) object with `S2N_PSK_HMAC_SHA256` as the default 
+ * PSK hash algorithm. An external PSK is a key established outside of TLS using a secure mutually agreed upon mechanism.
+ * 
+ * Use `s2n_psk_free` to free the memory allocated to the s2n external PSK object created by this API. 
+ *
+ * @return struct s2n_psk* Returns a pointer to the newly created external PSK object.
+ */
```

Comment:
We don't generally give a type here, and it looks a little weird with the C struct language. I think the comment is enough.